libs/pilight/protocols/433.92/hideki.c

```diff
@@ -0,0 +1,492 @@
+/*
+	Copyright (C) 2019 Falk Nedwal
+
+	This file is part of pilight.
+	This Source Code Form is subject to the terms of the Mozilla Public
+	License, v. 2.0. If a copy of the MPL was not distributed with this
+	file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+/*
+ *  this implementation is based on RTL_433 project code for HIDEKI
+ * */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "../../core/pilight.h"
+#include "../../core/common.h"
+#include "../../core/dso.h"
+#include "../../core/log.h"
+#include "../protocol.h"
+#include "../../core/binary.h"
+#include "../../core/gc.h"
+#include "hideki.h"
+
+#define MIN_PULSE_LENGTH 220
+#define MAX_PULSE_LENGTH 280
+
+#define MIN_RAW_LENGTH 130
+#define MAX_RAW_LENGTH 180
+#define PULSE_HIDEKI_WEATHER_LOWER 100	// experimentally detected
+#define PULSE_HIDEKI_WEATHER_UPPER 700	// experimentally detected
+
+#define HIDEKI_MAX_BYTES_PER_ROW 14
+
+#define CALIBRATION_COUNT 3
+
+//#define HIDEKI_DEBUG
+
+/*
+ * The received bits are inverted.
+ * Every 8 bits are stuffed with a (even) parity bit.
+ * The payload (excluding the header) has an byte parity (XOR) check
+ * The payload (excluding the header) has CRC-8, poly 0x07 init 0x00 check
+ * The payload bytes are reflected (LSB first / LSB last) after the CRC check
+ *
+ *    11111001 0  11110101 0  01110011 1 01111010 1  11001100 0  01000011 1  01000110 1  00111111 0  00001001 0  00010111 0
+ *    SYNC+HEAD P   RC cha P     LEN   P     Nr.? P   .1° 1°  P   10°  BV P   1%  10% P     ?     P     XOR   P     CRC   P
+ *
+ * TS04:
+ *    00000000  11111111  22222222  33333333  44444444  55555555  66666666  77777777  88888888 99999999
+ *    SYNC+HEAD cha   RC     LEN        Nr.?    1° .1°  VB   10°   10%  1%     ?         XOR      CRC
+ *
+ * Wind:
+ *    00000000  11111111  22222222  33333333  44444444  55555555  66666666  77777777  88888888 99999999 AAAAAAAA BBBBBBBB CCCCCCCC DDDDDDDD
+ *    SYNC+HEAD cha   RC     LEN        Nr.?    1° .1°  VB   10°    1° .1°  VB   10°   1W .1W  .1G 10W   10G 1G    w°  AA    XOR      CRC
+ *
+ * Rain:
+ *    00000000  11111111  22222222  33333333  44444444  55555555  66666666  77777777  88888888
+ *    SYNC+HEAD cha   RC   B LEN        Nr.?   RAIN_L    RAIN_H     0x66       XOR       CRC
+ *
+ */
+
+enum sensortypes { HIDEKI_UNKNOWN, HIDEKI_TEMP, HIDEKI_TS04, HIDEKI_WIND, HIDEKI_RAIN };
```

Comment:
Can you make this static?

Reply:
why that? 
I just tries it out and the compiler responded with:
/usr/src/pilight/libs/pilight/protocols/433.92/hideki.c:66:82: warning: useless storage class specifier in empty declaration
 static enum sensortypes { HIDEKI_UNKNOWN, HIDEKI_TEMP, HIDEKI_TS04, HIDEKI_WIND, HIDEKI_RAIN };

Reply:
Oeps, enums are always static. My bad.